src/connector/src/parser/mod.rs

```diff
@@ -488,6 +488,16 @@ impl SourceStreamChunkRowWriter<'_> {
                             .map(|ele| ScalarRefImpl::Utf8(ele.split_id)),
                     ));
                 }
+                (_, &Some(AdditionalColumnType::Payload(_))) => {
+                    // Get the whole payload as a single column
+                    // The frontend check guarantees that row encode must be `SourceEncode::Json`
+                    // here fake a column named "." to represent the whole payload
+                    // see the json accessor hack in `impl Access for JsonAccess<'_>`
+                    let mut desc_mock = desc.clone();
+                    desc_mock.name = ".".to_string();
+                    desc_mock.additional_column.column_type = None;
+                    parse_field(&desc_mock)
```

Comment:
Can we match against `AdditionalColumnType::Payload` here and access with an empty path to avoid the need for this hack?

https://github.com/risingwavelabs/risingwave/blob/11f7e617a4e1fcc2b0ab968e24aab0ff4a62075e/src/connector/src/parser/unified/kv_event.rs#L81-L82

Reply:
True bro, we can avoid the hack here. But we still need to hack on the `Access` trait level because we only allow access to inner fields. And it is not worthy to add a new function on the trait level.

Reply:
Can you elaborate more? What changes are needed for `Access` trait?

Reply:
> Can you elaborate more? What changes are needed for `Access` trait?

In the current impl, `Access` trait only has one func with signature `fn access<'a>(&'a self, path: &[&str], type_expected: &DataType) -> AccessResult<DatumCow<'a>>`. 
It fetches inner fields by path, but we never define a path to get the payload as a whole. Compared with adding a new func to the trait, I define `path = &["."]` for this case.

Reply:
What about simply using an empty slice for `path`?

Reply:
seems can avoid collision with json with key `"."`. fix later